oak-search/src/main/java/org/apache/jackrabbit/oak/plugins/index/search/update/TimedRefreshPolicy.java

```diff
@@ -53,16 +54,20 @@ public void refreshOnWriteIfRequired(Runnable refreshCallback) {
 
     @Override
     public void updated() {
-        dirty.set(true);
+        synchronized (lock) {
+            dirty.set(true);
+        }
     }
 
     private void refreshIfRequired(Runnable refreshCallback) {
-        if (dirty.get()){
-            long currentTime = clock.getTime();
-            if (currentTime - lastRefreshTime > refreshDelta
-                    && dirty.compareAndSet(true, false)){
-                lastRefreshTime = currentTime;
-                refreshCallback.run();
+        synchronized (lock) {
```

Comment:
Could you please explain what is the intention behind synchronizing on `lock` . Do we want that dirty stays `false` while `refreshCallback` is running?

Reply:
I think it's not needed to synchronize updated(). It should be enough to synchronize just the call to refreshCallback.run() in my view.